app/src/main/java/com/tyranor/next/core/game/storage/GameLibraryRepository.kt

```
@@ -350,7 +355,10 @@ object GameLibraryRepository {
                 writeMutex.withLock { block(app) }
                 schedulePrefsMirror(app)
             } catch (t: Throwable) {
+                if (t is kotlinx.coroutines.CancellationException) throw t
```

detekt (1.23.8)

[warning] Instead of catching for a general exception type and checking for a specific exception type, use multiple catch blocks.

(detekt.exceptions.InstanceOfCheckForException)